src/callbacks_step/timeseries_dg2d.jl

```diff
@@ -0,0 +1,139 @@
+
+# Creates cache for time series callback
+function create_cache_timeseries(point_coordinates, mesh::TreeMesh{2}, dg, cache)
+  # Determine element ids for point coordinates
+  element_ids = get_elements_by_coordinates(point_coordinates, mesh, dg, cache)
+
+  # Calculate & store Lagrange interpolation polynomials
+  interpolating_polynomials = calc_interpolating_polynomials(point_coordinates, element_ids, mesh,
+                                                             dg, cache)
+
+  timeseries_cache = (; element_ids, interpolating_polynomials)
+
+  return timeseries_cache
+end
+
+
+# Find element ids containing coordinates given as a matrix [ndims, npoints]
+function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh, dg, cache)
+  @assert length(element_ids) == size(coordinates, 2) "size mismatch"
+
+  @unpack cell_ids = cache.elements
+  @unpack tree = mesh
+
+  # Reset element ids - 0 indicates "not (yet) found"
+  element_ids .= 0
+  found_elements = 0
+
+  # Iterate over all elements
+  for element in eachelement(dg, cache)
+    # Get cell id
+    cell_id = cell_ids[element]
+
+    # Iterate over coordinates
+    for index in 1:length(element_ids)
+      # Skip coordinates for which an element has already been found
+      if element_ids[index] > 0
+        continue
+      end
+
+      # Construct point
+      x = SVector(ntuple(i -> coordinates[i, index], ndims(mesh)))
```

Comment:
Why don't you store the coordinates as a vector of `SVector`s if you just use them like this?

Reply:
Okay